stix_shifter/stix_transmission/src/modules/aws_cloud_watch_logs/aws_cloud_watch_logs_query_connector.py

```diff
@@ -0,0 +1,47 @@
+from ..base.base_query_connector import BaseQueryConnector
+from .....utils.error_response import ErrorResponder
+import json
+
+
+class AWSCloudWatchLogsQueryConnector(BaseQueryConnector):
+
+    def __init__(self, client, log_group_names):
+        self.client = client
+        self.log_group_names = log_group_names
+
+    def create_query_connection(self, query):
+        """
+        Function to create query connection
+        :param query: str, Query
+        :return: dict
+        """
+        return_obj = dict()
+        response_dict = dict()
+        try:
+            query = json.loads(query)
+            is_logtypepresent = False
+            if self.log_group_names:
+                if isinstance(self.log_group_names, dict):
```

Comment:
Needs some explanation about below block of code where you are checking logType and creating a list of logGroupNames. first of all, not sure why you would need to specify the log types such as `guradduty`/`vpcflow` as a key to create a list of log group names-
`"log_group_names":{"guardduty":["/aws/events/guardduty"]}`

I understand, you are trying to specify what kind of logs contains in that log group. But can the user just mention a log group or a list of log groups without mentioning what kind of logs are present in there? Because I think multiple types of logs can contain in a single log group. 
I think we should just set the list of log group names in connection object like this-
`"log_group_names":["/aws/events/guardduty"., "USEast1_FlowLogs"]`

In that way, you may not need to do all of these below checks. You can just directly push the logGroupNames list into the query object.
Let me know if you have any different opinion.

Reply:
As per above suggestion, if loggroupnames are given without logtype, each query will search for the pattern in all the loggroups eventhough the group is not containing related logs. This may have performance degrade. Inorder to handle this, we have supported more granular level by adding logtypes.
We have made new change in the above code by adding another logtype says "default". This "default" logtype keyword, takes list of groupnames as input.  Here loggroups will be added to query if specific logtype available in loggroupnames like 'vpcflow, guardduty' else loggroups in "default" type will be added. 
Examples:
Input case1: "log_group_names": {"guardduty": ["/aws/events/guardduty"], "vpcflow":"USEast1_FlowLogs", "default":["/aws/events/guardduty", "USEast1_FlowLogs"]} 

In Input case1, we have "guardduty" logtype in input, hence guardduty query will be appended with "/aws/events/guardduty" loggroup.
We have "vpcflow" logtype in input, hence vpcflow query will be appended with "USEast1_FlowLogs"
"default" type will be ignored.

Input case2: "log_group_names": {"guardduty": ["/aws/events/guardduty"], "default":["/aws/events/guardduty", "USEast1_FlowLogs"]} 

In Input case2, we have "guardduty" logtype in input, so guardduty query will be appended with "/aws/events/guardduty" loggroup
we don't have "vpcflow" logtype in input, so vpcflow query will be appended with "default" type loggroups- ["/aws/events/guardduty", "USEast1_FlowLogs"]

Input case3: "log_group_names": {"default":["/aws/events/guardduty", "USEast1_FlowLogs"]} 

In Input case3, we don't have "guardduty" and "vpcflow" logtype in input, so both the queries will be appended  with "default" type loggroups- ["/aws/events/guardduty", "USEast1_FlowLogs"]. This case will satisfy your suggestion.
Please provide your comments on this.

Reply:
> As per above suggestion, if loggroupnames are given without logtype, each query will search for the pattern in all the loggroups eventhough the group is not containing related logs. This may have performance degrade. Inorder to handle this, we have supported more granular level by adding logtypes.

Not sure what you are trying to explain. if someone provide a loggroup name without a logtype than why all the log groups will be queried? the purpose of specifying a log group is just querying a individual log group no matter what type of logs are stored in there, am i correct? User will just provide a single or multiple log group name without knowing what type of logs are there. we are simply just going to query that log group and bring back the logs.

Reply:
I have sent separate mail thread with additional details. Please let me know your comments.